### ISSUE TITLE
BEM's CSS design rules do not work properly with class-naming rules

### ISSUE THREAD

**Issue by mmrakt** [markuplint/markuplint#1263]
- Markuplint version: `3.14.0`
- Parser lang: HTML
- Node.js version: `18.18.2`
- OS: macOS

## Describe the bug

I tried applying BEM's CSS design rules based on the [official guide](https://markuplint.dev/docs/rules/class-naming#configuration-example) with the class-naming rule, but it doesn't seem to work correctly because some attributes are recognized as blocks.

`.markuplintrc.js`

```js
module.exports = {
  extends: ['markuplint:recommended'],
  rules: {
    'class-naming': '/.+/',
  },
  childNodeRules: [
    {
      regexSelector: {
        attrName: 'class',
        attrValue: '/^(?<BlockName>[A-Z][a-z0-9]+)(?:__[a-z][a-z0-9-]+)?$/',
      },
      rules: {
        'class-naming': {
          value: ['/^{{BlockName}}__[a-z][a-z0-9-]+$/', '/^([A-Z][a-z0-9]+)$/'],
        },
      },
    },
  ],
};
```

Also, if I define the class attribute last, it works correctly.

## Code Example

```html
<div class="carousel">
    <div class="carousel__slides" aria-live="off" >
	    <div class="carousel__slide">
		<p class="carousel__label">slide 1</p>
	    </div>
    </div>
</div>
```

## Steps To Reproduce

Run markuplint over the above code.

## The current behavior

following  warning is given.

```
 warning: The "carousel__slide" class name is unmatched with the below patterns: "/^off__[a-z][a-z0-9-]+$/", "/^([A-Z][a-z0-9]+)$/" (class-naming)
```

## The expected behavior

No warning  is given.

**Comment by YusukeHirao** [comment]
@mmrakt 

In this case, the naming convention for blocks here starts with an upper-case letter. This might have caused some instances to go undetected. Could you please try changing `carousel` to `Carousel`?

**Comment by mmrakt** [comment]
@YusukeHirao 
Sorry for the late reply.

> Could you please try changing carousel to Carousel?



I tried to correct the problem as you suggested, but the result was the same.


```html
<div class="Carousel">
    <div class="Carousel__slides" aria-live="off" >
	    <div class="Carousel__slide">
		<p class="Carousel__label">slide 1</p>
	    </div>
    </div>
</div>
```


```
warning: The "Carousel__slide" class name is unmatched with the below patterns: "/^off__[a-z][a-z0-9-]+$/", "/^([A-Z][a-z0-9]+)$/" (class-naming)
```
